ocean_lib/assets/asset.py

```diff
@@ -163,16 +163,41 @@ def add_service(self, service: Service) -> None:
 
     @enforce_types
     def get_service_by_id(self, service_id: str) -> Service:
-        """Return the Service with the matching id"""
-        return next((service for service in self.services if service.id == service_id))
+        """Return Service with the given id.
+        Return None if service with the given id not found."""
+        return next(
+            (service for service in self.services if service.id == service_id), None
+        )
+
+    @enforce_types
+    def get_service_by_index(self, service_index: int) -> Service:
+        """Return Service with the given index.
+        Return None if service with the given index not found."""
+        return (
+            self.services[service_index] if service_index < len(self.services) else None
+        )
 
     @enforce_types
     def get_service(self, service_type: str) -> Optional[Service]:
-        """Return the first Service with the given service type."""
+        """Return first Service with the given service type.
+        Return None if service with the given service type not found."""
         return next(
             (service for service in self.services if service.type == service_type), None
         )
 
+    @enforce_types
+    def get_service_index_by_id(self, service_id: str) -> int:
```

Comment:
I like this helper, but I really don't like this function's name. I would suggest the parameter to be a service, not the service id, and let's name it something more suggestive. E.g. "get_index_of_service"?

This is very good, and it helped me find an error within the consume flows.

Reply:
The function exists on the marketplace-flow branch now, so you can copy it over.